src/main/host/syscall/clone.c

```diff
@@ -72,3 +72,8 @@ SysCallReturn syscallhandler_clone(SysCallHandler* sys, const SysCallArgs* args)
 
     return (SysCallReturn){.state = SYSCALL_DONE, .retval.as_i64 = thread_getID(child)};
 }
+
+SysCallReturn syscallhandler_gettid(SysCallHandler* sys, const SysCallArgs* args) {
+    utility_assert(sys && args);
+    return (SysCallReturn){.state = SYSCALL_DONE, .retval.as_i64 = thread_getID(sys->thread)};
+}
```

Comment:
You beat me to it! I also [implemented `gettid`](https://github.com/shadow/shadow/pull/982/commits/dc6e586333523ff092290555d8f38d6d0bc1918c), but I put mine in `unistd.c` (alongside `getpid`).

Reply:
I considered putting it there, but since it isn't actually in unistd.h thought it made more sense to put it with "thread stuff" than "get an ID stuff". Happy to move it if you think unistd.c makes more sense though.

Reply:
It's fine. I'll remove mine from unistd.

Reply:
`set_tid_address` is a thread thing and is in unistd though, so who knows.